docs: update v.debug docs - callstack+trace (#20854)
vlang/v: doc/docs.md

check if the current context is an anon function (`anon?`), a method (`method?`) 
or a generic method (`generic?`) and clear the terminal window (`clear`).

## Call stack

> Using also `v.debug` module you can have call stack on your code for debugging 

Reviewer: ```suggestion
Also using the `v.debug` module you can have a call stack on your code for debugging 
```

> Also using the `v.debug` module you can have a call stack on your code for debugging 
> purpose!

Reviewer 2: ```suggestion
Also using the `v.debug` module you show the call stack on your code while debugging.
```

> To enable such feature use `-d callstack` when running your code or building 
> source files, see below:

Reviewer 2: ```suggestion
To enable this feature use `-d callstack` when running your code or building 
source files:
```

```v
import v.debug

fn test(i int) {
	if i > 9 {
		debug.dump_callstack()
	}
}

fn do_something() {
	for i := 0; i <= 10; i++ {
		test(i)
	}
}

fn main() {
	do_something()
}
```

```
$ v -d callstack run example.v
Backtrace:
--------------------------------------------------
example.v:16   | > main.main
example.v:11   |  > main.do_something
example.v:5    |   > main.test
--------------------------------------------------
```

## Trace

> Another feature from `v.debug` is the possibility to add hook functions 
> before and after each function call on your code.

Reviewer 2: ```suggestion
Another feature of `v.debug` is the possibility to add hook functions 
before and after each function call.
```

> To enable such feature use `-d trace` when running your code or building 
> source files, see below:

Reviewer 2: ```suggestion
To enable this feature use `-d trace` when running your code or building 
source files:
```

```v
import v.debug

fn main() {
	hook1 := debug.add_before_call(fn (fn_name string) {
		println('> before ${fn_name}')
	})
	hook2 := debug.add_after_call(fn (fn_name string) {
		println('> after ${fn_name}')
	})
	anon := fn () {
		println('call')
	}
	anon()

> 	// optionall you can remove the hooks:

Reviewer 3: ```suggestion
	// optionally you can remove the hooks:
```
